### PR TITLE
[Fix] Drop the shadowed copies of the lm_head quant-method helpers

### PR DESCRIPTION
**Pull request by JasonKeyiL**
## Motivation

`logits_processor.py` defines `_has_lm_head_runtime_attrs` and `should_apply_lm_head_quant_method` twice at module level, at lines 123/127 and 1230/1234. Python binds the later definition, so the pair near the top of the file has never run.

The two copies are no longer the same. The duplication came in with the rebase in #35758 (2026-08-28), which left the file with two copies of each. Three days later #35120 added the NVFP4 W4A16 branch to the copy Python actually binds, and only to that one:

```python
 if method_name == "ModelOptFp4LinearMethod":
+    if quant_method.quant_mode == "w4a16":
+        return lm_head.weight.dtype == torch.uint8 and _has_lm_head_runtime_attrs(...)
```

So the shadowed copy answers `False` for a w4a16 lm_head that the live one accepts. This is the predicate that decides whether a draft model may reuse a quantized target lm_head, and reading the top of the file currently gives the wrong answer about which layouts are supported.

`ast.unparse` of the two `_has_lm_head_runtime_attrs` bodies is identical, so that one is pure duplication.

## Modifications

Remove the shadowed definitions at 123 and 127. Nothing else changes.

## Accuracy Tests

No behaviour change, and there is nothing new to test: the definitions that survive are the ones Python was already binding.

Verified rather than assumed:

```
_has_lm_head_runtime_attrs:         ast.unparse(kept) == ast.unparse(main's last def)  -> True
should_apply_lm_head_quant_method:  ast.unparse(kept) == ast.unparse(main's last def)  -> True
module top-level symbols:           41 -> 39   (exactly the two removed)
```

`test/registered/unit/spec/test_eagle_draft_extend_logits.py` passes locally (5 tests), and `should_apply_lm_head_quant_method` still imports and evaluates as before. `black --check` and `isort --check-only` are clean.

Verified on macOS/arm64, Python 3.13, CPU only.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).

---
### CI States

Latest PR Test (Base): :x: [Run #35055132838](https://github.com/sgl-project/sglang/actions/runs/35055132838)
Latest PR Test (Extra): :x: [Run #35055132737](https://github.com/sgl-project/sglang/actions/runs/35055132737)
Latest PR Test (AMD ROCm 10): :x: [Run #35055132762](https://github.com/sgl-project/sglang/actions/runs/35055132762)

### CODE REVIEW

**Comment by JasonKeyiL** [comment]
Could someone add the `run-ci` label? `/tag-run-ci-label` is a no-op for me, so `pr-gate` blocks this.

It is a pure deletion of two module-level definitions Python never binds; `ast.unparse` of what survives matches what `main` already executes, so there is no behaviour to test.
